warp-element/src/utils.js

```diff
@@ -1,63 +1,80 @@
-import { unsafeCSS } from 'lit';
+import { unsafeCSS } from "lit";
 
 const isServer = () => {
-    return !(typeof window !== 'undefined');
+  return !(typeof window !== "undefined");
 };
 
-const parseBrand = (str = '') => {
-    if (str.toLocaleLowerCase().includes('blocket')) return { sld: 'blocket', tld: 'se' };
-    if (str.toLocaleLowerCase().includes('tori')) return { sld: 'tori', tld: 'fi' };
-    if (str.toLocaleLowerCase().includes('finn')) return { sld: 'finn', tld: 'no' };
-    if (str.toLocaleLowerCase().includes('dba')) return { sld: 'dba', tld: 'dk' };
-    return { sld: 'finn', tld: 'no' };
+const parseBrand = (str = "") => {
+  if (str.toLocaleLowerCase().includes("blocket"))
+    return { sld: "blocket", tld: "se" };
+  if (str.toLocaleLowerCase().includes("tori"))
+    return { sld: "tori", tld: "fi" };
+  if (str.toLocaleLowerCase().includes("finn"))
+    return { sld: "finn", tld: "no" };
+  if (str.toLocaleLowerCase().includes("dba")) return { sld: "dba", tld: "dk" };
+  return { sld: "finn", tld: "no" };
 };
 
-
 /*
  * @param {string} [brandStr] - a string with brand information
  * @returns {import("./global.js").Brand} brand object
  */
-export const getBrand = (brandStr = '') => {
-    if (brandStr !== '') return parseBrand(brandStr);
-    if (!isServer() && window?.location?.host) return parseBrand(window.location.host);
-    if (isServer() && process?.env?.NMP_BRAND) return parseBrand(process.env.NMP_BRAND);
-    return parseBrand();
+export const getBrand = (brandStr = "") => {
+  if (brandStr !== "") return parseBrand(brandStr);
+  if (!isServer() && window?.location?.host)
+    return parseBrand(window.location.host);
+  if (isServer() && process?.env?.NMP_BRAND)
+    return parseBrand(process.env.NMP_BRAND);
+  return parseBrand();
 };
 
 const loadStyles = async (urls = []) => {
-    const server = isServer();
-
-    if (server) {
-        return urls.map((style) => {
-            return unsafeCSS(`@import url('${style}');`);
-        });
-    }
-
-    const requests = await Promise.all(urls.map((url) => {
-        return fetch(url);
-    }));
-
-    const sheets = await Promise.all(requests.map((response) => {
-        return response.text();
-    }));
-
-    return sheets.map((style) => {
-        const sheet = new CSSStyleSheet();
-        sheet.replaceSync(style);
-        return sheet;
+  const server = isServer();
+
+  if (server) {
+    return urls.map((style) => {
+      return unsafeCSS(`@import url('${style}');`);
     });
+  }
+
+  // only load polyfill if needed, and only client side.
+  const supportsAdoptingStyleSheets =
+    "adoptedStyleSheets" in Document.prototype &&
+    "replace" in CSSStyleSheet.prototype;
+
+  if (!supportsAdoptingStyleSheets) {
+    await import("construct-style-sheets-polyfill");
```

Comment:
How will the browser load this? I think we need to keep the polyfill as a separate file and only load it when needed. I don't think we can keep this as a bare import.

Reply:
In the same way it loads lit currently. Either the consuming package inlines it or import maps it to an absolute url with Eik.

Reply:
For simplicity I'll publish a version of construct-style-sheets-polyfill to Eik and we can reference that directly here.